pytorch/containers/hp_tuning/README.md

```diff
@@ -0,0 +1,183 @@
+* [Install pandas](https://pandas.pydata.org/pandas-docs/stable/install.html) [Optional: used if running locally]
+
+These variables will be needed for the following steps.
+
+**Replace these variables:**
+```
+# PROJECT_ID: your project's id. Use the PROJECT_ID that matches your Google Cloud Platform project.
+export PROJECT_ID=YOUR_PROJECT_ID
+
+# BUCKET_ID: the bucket id you created above.
+export BUCKET_ID=BUCKET_ID
+```
+
+Additional variables:
+```
+# JOB_DIR: with the path to a Google Cloud Storage location to use for job output.
+export JOB_DIR=gs://YOUR_BUCKET_ID/hp_tuning
+
+# IMAGE_REPO_NAME: where the image will be stored on Cloud Container Registry
+export IMAGE_REPO_NAME=sonar_hp_tuning_pytorch_container
+
+# IMAGE_TAG: a easily identifiable tag for your docker image
+export IMAGE_TAG=sonar_hp_tuning_pytorch
+
+# IMAGE_URI: the complete URI location for Cloud Container Registry
+export IMAGE_URI=gcr.io/$PROJECT_ID/$IMAGE_REPO_NAME:$IMAGE_TAG
+
+# REGION: select a region from https://cloud.google.com/ml-engine/docs/regions
+# or use the default '`us-central1`'. The region is where the model will be deployed.
+export REGION=us-central1
+
+# JOB_NAME: the name of your job running on Cloud ML Engine.
+export JOB_NAME=hp_tuning_container_job_$(date +%Y%m%d_%H%M%S)
+```
+
+# Part 1: Create the model you want to train
+Here we provide an example [model.py](model.py) that trains a PyTorch model
+to predict whether the given sonar signals are bouncing off a metal cylinder or off a cylindrical
+rock. You will also find the code that handles argument parsing for the hyperparameter values as
+well as the code to track the performance for hptuning.
+
+Open up the [task.py](task.py) to see exactly how the model is called during
+training.
+
+[data_utils.py](data_utils.py) is used to download / load the data
+and exports your trained model and uploads the model to Google Cloud Storage.
+
+The dataset for the model is hosted originally at the UCI Machine Learning Repository. We've 
+[hosted the sonar dataset in Cloud Storage](https://storage.cloud.google.com/cloud-samples-data/ml-engine/sonar/sonar.all-data?organizationId=433637338589&_ga=2.163217084.-1279615720.1534888758)
+for use with this sample.
+
+# Part 2: Create the docker Image
+Open the [Dockerfile](Dockerfile) to see how the Docker image is created that will run on Cloud
+ML Engine.
+
+# Part 3: Build the docker Image
+```
+docker build -f Dockerfile -t $IMAGE_URI ./
+```
+
+# Part 4: Test your docker image locally
+```
+docker run $IMAGE_URI --epochs 1
+```
+
+If it ran successfully, the output should look similar to: `Accuracy: 58%`.
+
+# Part 5: Deploy the docker image to Cloud Container Registry
+You should have configured docker to use Cloud Container Registry, found
+[here](https://cloud.google.com/container-registry/docs/pushing-and-pulling).
+```
+docker push $IMAGE_URI
+```
+
+# Part 6: Submit your training job
+Creates a hptuning config file for your job request.
```

Comment:
add link to docs explaining how hptuning on CMLE works.

Reply:
I have up at the top an overview of HP Tuning links under pre-req.